src/bootstrap/src/lib.rs

```diff
@@ -69,16 +69,61 @@ const LLVM_TOOLS: &[&str] = &[
 /// LLD file names for all flavors.
 const LLD_FILE_NAMES: &[&str] = &["ld.lld", "ld64.lld", "lld-link", "wasm-ld"];
 
+#[derive(Clone, Debug)]
+pub struct ConfigChange {
+    /// Represents the ID of PR caused major change on bootstrap.
+    pub change_id: usize,
+    pub severity: Severity,
+    /// Provides a short summary of the change that will guide developers
+    /// on "how to handle/behave" in response to the changes.
+    pub summary: &'static str,
+}
+
+#[derive(Clone, Debug)]
+pub enum Severity {
+    /// Used when upstream bootstrap configurations continue working as before.
+    Info,
+    /// Used when upstream bootstrap configurations change, and it may
+    /// require developers to update their build configuration.
```

Comment:
What are "upstream bootstrap configurations"?

Reply:
build configurations / config.toml. Will make it more clear.

Reply:
Is that the same as "profiles"?

Reply:
No, profiles doesn't include/cover the entire `config.toml` options. But `config.toml` covers profiles too.

Reply:
> Used when upstream bootstrap configurations change

e.g., changing the default value of any option in `config.example.toml`, or directly in the bootstrap configuration deserializer.

Reply:
So that sounds like "Used when default values change"?

Reply:
"Used when default values change or options removed in build configurations."

What about that? I think "or options removed" is worth to mention.

Reply:
Yeah sure. Though "removed in build configurations" is again unclear, since you didn't remove them in the user's config. What about

"Used when the default value of an option changes, or support for an option is removed entirely"

Reply:
https://github.com/rust-lang/rust/pull/117815/commits/06f6cd95beca124a54a7346ea2e31b75e1bcfc5a